hack/tools/tilt-prepare/main.go

```diff
@@ -249,19 +261,76 @@ func tiltResources(ctx context.Context, ts *tiltSettings) error {
 	}
 
 	// Add a provider task for each name/path defined using the --provider flag.
-	for _, p := range *providersBuildsFlag {
-		pValues := strings.Split(p, ":")
-		if len(pValues) != 2 {
-			return errors.Errorf("[resources] failed to parse --provider flag %s: value should be in the form of name:path", p)
+	for _, p := range ts.ProviderBuilds {
+		name, path := providerContext(p)
+		tasks[name] = providerTask(name, fmt.Sprintf("%s/config/default", path), ts)
+	}
+
+	// Add a task for the provider repos defined by the provider_repo key in tilt file
+	for _, p := range ts.ProviderRepos {
+		name, path, err := loadProviders(p)
+		if err != nil {
+			return err
 		}
-		name := pValues[0]
-		path := pValues[1]
 		tasks[name] = providerTask(name, fmt.Sprintf("%s/config/default", path), ts)
 	}
 
 	return runTaskGroup(ctx, "resources", tasks)
 }
 
+func providerContext(s string) (string, string) {
+	var path string
+	switch s {
+	case "core":
+		path = "."
+	case "kubeadm-bootstrap":
+		path = "bootstrap/kubeadm"
+	case "kubeadm-control-plane":
+		path = "controlplane/kubeadm"
+	case "docker":
+		path = "test/infrastructure/docker"
+	}
+	return s, path
+}
+
+func loadProviders(r string) (string, string, error) {
+	var contextPath string
+	providerData, err := readProviderSettings(r)
+	if err != nil {
+		return "", "", err
+	}
+	if providerData.Context != nil {
+		contextPath = r + "/" + *providerData.Context
+	} else {
+		contextPath = r
+	}
+	return providerData.Name, contextPath, nil
+}
+
+func readProviderSettings(path string) (*providerSettings, error) {
+	path, err := checkProviderFileFormat(path)
+	if err != nil {
+		return nil, err
+	}
+
+	providerData := &providerSettings{}
+	if err := yaml.Unmarshal([]byte(path), &providerData); err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("failed to read provider path %s tilt settings", path))
+	}
+
+	return providerData, nil
+}
```

Comment:
```suggestion
func readProviderSettings(path string) ([]providerSettings, error) {
	path, err := checkProviderFileFormat(path)
	if err != nil {
		return nil, err
	}

	content, err := os.ReadFile(path)
	if err != nil {
		return nil, err
	}

	ps := []providerSettings{}
	# Check if content contains an array.
	if strings.HasPrefix(string(content), "[") || strings.HasPrefix(string(content), "-") {
		if err := yaml.Unmarshal(content, &ps); err != nil {
			return nil, errors.Wrap(err, fmt.Sprintf("failed to read provider path %s tilt settings", path))
		}
	} else {
		p := providerSettings{}
		if err := yaml.Unmarshal(content, &p); err != nil {
			return nil, errors.Wrap(err, fmt.Sprintf("failed to read provider path %s tilt settings", path))
		}
		ps = append(ps, p)
	}

	return ps, nil
}
```

1. we have to read the file
2. provider settings **can be** an array (if there's  a better way to detect the array then my bad hack, please let's use that)

=> Please handle the array one level above in `loadProviders` accordingly and return a `map[string]string` (providerName => context) there
=> Then handle the map in tiltResources accordingly by adding all map entries to providerPaths

Reply:
Please test if everything works with e.g. a config like this:
```yaml
provider_repos:
- "../cluster-api-provider-aws"
- "../cluster-api-provider-vsphere"
enable_providers:
- docker
- kubeadm-bootstrap
- kubeadm-control-plane
- aws
- vsphere
```